src/namespace/privilegednamespace.js

```diff
@@ -287,4 +362,165 @@ export class PrivilegedNamespace extends CustomNamespace {
+        }
+
+        logger.info(`User (${socket.username}) banned the device with uuid ${uuid}`);
+        callback(true);
+    };
+
+    /**
+     * Unban a user from all admin future debates
+     * unbanObj contains the required information (uuid)
+     */
+    unbanUser = (socket) => async (unbanObj, callback) => {
+        logger.debug(`unbanUser received from user (${socket.username}), id(${socket.id})`);
+
+        if (!TypeCheck.isFunction(callback)) {
+            logger.debug(`callback is not a function.`);
+            return;
+        }
+
+        let {uuid} = unbanObj;
+        if (!TypeCheck.isString(uuid)) {
+            logger.debug('Invalid arguments for unbanUser');
+            callback(false);
+            return;
+        }
+
+        // Check if the device is banned by this user
+        let isBanned = await dbManager.isDeviceBanned(uuid, socket.username);
+        if (!isBanned) {
+            logger.info(`Device with uuid (${uuid}) is not banned`);
+            callback(true);
+            return;
+        }
+
+        // Ban the device in the database
+        let res = await dbManager.unbanDevice(uuid, socket.username);
+        if (res === false) { // We know the device exists and was banned by this specific user...
+            logger.error(`Couldn't unban device with uuid ${uuid}`);
+            callback(false);
+            return;
+        }
+
+        logger.info(`User (${socket.username}) unbanned the device with uuid ${uuid}`);
+        callback(true);
+    };
+
+    /**
+     * Approve a suggestion with the specified id and debate
+     * approveObj contains the required information (debateId and suggestionId)
+     */
+    approveQuestion = (socket) => (approveObj, callback) => {
+        logger.debug(`approveQuestion received from user (${socket.username}), id(${socket.id})`);
+
+        if (!TypeCheck.isFunction(callback)) {
+            logger.debug(`callback is not a function.`);
+            return;
+        }
+
+        let {suggestionId, debateId} = approveObj;
+        if (!TypeCheck.isInteger(suggestionId) || !TypeCheck.isInteger(debateId)) {
+            logger.debug('Invalid arguments for approveSuggestion');
+            callback(false);
+            return;
+        }
+
+        const debate = this.getActiveDebate(debateId);
+        if (debate == null) {
+            logger.debug(`Debate with id (${debateId}) not found.`);
+            callback(false);
+            return;
+        }
+
+        const res = debate.questionSuggestion.approveSuggestion(suggestionId);
+        if (res === false) {
+            logger.debug('Cannot approve suggestion.');
+            callback(false);
+            return;
+        }
+
+        logger.info(`User (${socket.username}) approved suggestion with id (${suggestionId})`);
+        callback(true);
+    };
+
+    /**
+     * Reject a suggestion with the specified id and debate
+     * rejectObj contains the required information (debateId and suggestionId)
```

Comment:
Same comment here. `debateId` and `suggestionId` are not in order compared to ligne 502 `let {suggestionId, debateId} = rejectObj`.

Reply:
Order doesn't matter as it is an object. I still updated it for readability in 2cab5a0.